src/Discord.Net.Rest/Entities/Messages/MessageHelper.cs

```diff
@@ -46,13 +46,39 @@ internal static class MessageHelper
             await client.ApiClient.RemoveAllReactionsAsync(msg.Channel.Id, msg.Id, options);
         }
 
-        public static async Task<IReadOnlyCollection<IUser>> GetReactionUsersAsync(IMessage msg, IEmote emote,
+        public static IAsyncEnumerable<IReadOnlyCollection<IUser>> GetReactionUsersAsync(IMessage msg, IEmote emote,
             Action<GetReactionUsersParams> func, BaseDiscordClient client, RequestOptions options)
         {
             var args = new GetReactionUsersParams();
             func(args);
-            string emoji = (emote is Emote e ? $"{e.Name}:{e.Id}" : emote.Name);
-            return (await client.ApiClient.GetReactionUsersAsync(msg.Channel.Id, msg.Id, emoji, args, options).ConfigureAwait(false)).Select(u => RestUser.Create(client, u)).ToImmutableArray();
+
+            var emoji = (emote is Emote e ? $"{e.Name}:{e.Id}" : emote.Name);
```

Comment:
Better check for nulls before (msg and emote).
This will throw if emote is null.

Reply:
The message shouldn't be null, since the whole class is internal, and the places where it is called (Message objects) can't be null else Nullrefs will be thrown before it enters that method. So is it strictly necessary?

Reply:
Or is it standard throughout the rest of the library?

Reply:
You do have a point about the message, but `emote` could still receive null and throw.